- Added manual-lice-register.js file

melyav92 opened the pull request:
- added manual-lice-register.spec.js file
- added page objects
- added test for adding new manual lice count report by using 'Add new button'

melyav92 committed:
- Added manual-lice-register.js file
- added manual-lice-register.spec.js file
- added page objects
- added test for adding new manual lice count report by using 'Add new button'

tamysko commented on cypress/integration/tests/manual-lice-register.spec.js:
```diff
@@ -0,0 +1,80 @@
+
+const locationId = 2127;
+const currentYear = Cypress.moment( ).format("YYYY");
+const reportDate = Cypress.moment().format("DD/MM/YYYY");
```
Please move this to common helpers. It should be vars, not const.

tamysko commented on cypress/integration/tests/manual-lice-register.spec.js:
```diff
@@ -0,0 +1,80 @@
+const currentYear = Cypress.moment( ).format("YYYY");
+const reportDate = Cypress.moment().format("DD/MM/YYYY");
+const username = 'bolacslu';
+const password = '123456';
```
login and pwd should be read from config file as default, in the test they could be rewritten to a new ones, but do not hard code them here. this is a good solution.

tamysko commented on cypress/integration/tests/manual-lice-register.spec.js:
```diff
@@ -0,0 +1,80 @@
+const calligusForPenM2 = '2';
+const fishCountForPenM2 = '10';
+const penM2Comment = "Comment for pen M2";
+const successfulToasterPopupMessage = 'Lice count was saved successfully';
```
This should be part of the page object file. cypress/support/pageobjects/manual-lice-register.js

tamysko commented on cypress/support/pageobjects/manual-lice-register.js:
```diff
@@ -0,0 +1,90 @@
+        return this.getPenObjectByName(penName)
+            .parents('tr')
+            .find('[data-type="comments"]')
+    };
```
It is not adding comments, it is finding input. I suggest renaming variable to something more suitable. 
e.g. commentForPenInput

tamysko commented on cypress/support/pageobjects/manual-lice-register.js:
```diff
@@ -0,0 +1,90 @@
+    };
+    countedDateAboveTheTable(){
+       return cy.get('#scp-lice-count-date-value')
+    }
```
Not the best method name. it is not clear what it is.

tamysko commented on cypress/support/pageobjects/manual-lice-register.js:
```diff
@@ -0,0 +1,90 @@
+        cy.request({
+            method: 'DELETE',
+            url: `api/lice-count/delete?locationId=${locationId}&date=${deleteReportDate}`,
+            //failOnStatusCode: false
```
What is comment about?

tamysko commented on cypress/support/pageobjects/manual-lice-register.js:
```diff
@@ -0,0 +1,90 @@
+            .should('be.visible')
+    }
+    sendDeleteReportRequestForTheCurrentDate(locationId){
+        const deleteReportDate = Cypress.moment().format("YYYY-MM-DD");
```
Are you sure it's const?

tamysko commented on cypress/support/pageobjects/manual-lice-register.js:
```diff
@@ -0,0 +1,90 @@
+export class ManualLiceRegister{
+    loadingSpinner(){
+        return  cy.get('.scp-data-loading-text')
```
You have different loading spinner in harvests and lice-register. I think they should be consistent. Maybe it;s a bug in the system?)

tamysko commented on cypress/integration/tests/manual-lice-register.spec.js:
```diff
@@ -0,0 +1,80 @@
+        manualLiceRegister.sendDeleteReportRequestForTheCurrentDate(locationId)
+    })
+
+     */
```
What is this for?

tamysko commented on cypress/support/pageobjects/manual-lice-register.js:
```diff
@@ -0,0 +1,90 @@
+        return cy.get('.months-container')
+            .should('be.visible')
+    }
+    sendDeleteReportRequestForTheCurrentDate(locationId){
```
Why dont you make it generic?
sendDeleteReportRequest(locationId, date) {
body
}

tamysko commented on cypress/support/pageobjects/manual-lice-register.js:
```diff
@@ -0,0 +1,90 @@
+
+
+
+}
```
add empty line at the end :)

tamysko commented on cypress/integration/tests/manual-lice-register.spec.js:
```diff
@@ -0,0 +1,80 @@
+import {LoginPage} from "../../support/pageobjects/login-page";
+import {ManualLiceRegister} from "../../support/pageobjects/manual-lice-register";
+
+const locationId = 2127;
```
hardcoded id is not good at all. 
Why don't you set it up? or if it's something global like login, move it to config file?

melyav92 committed:
- updated objects names
- changed report date variable from 'const' to 'let'
- added spaces between objects
- refactored select current date object

melyav92 committed:
- removed date picker opening step

melyav92 committed:
- moved username, password, pen names, location id to the cypress.json file as env variables

melyav92 committed:
Merge branch 'master' into manual-lice-register.spec

melyav92 committed:
- moved delete report date variable out of send delete request object

melyav92 committed:
Merge remote-tracking branch 'origin/manual-lice-register.spec' into manual-lice-register.spec

tamysko reviewed (changes requested):
Please resolve all comments.

melyav92 committed:
- added fixtures file, moved test data to the fixtures file
- refactored delete report request
- removed current year variable
- refactored to use test data from fixtures

melyav92 committed:
- refactored select current date object

melyav92 committed:
- refactored getting fixtures data by using import
- added test that verifies updated data in the report

melyav92 committed:
- added verify updated data in the report
